contract/src/events/events_ces.rs

```diff
@@ -43,58 +40,65 @@ impl Burn {
 #[derive(Event, Debug, PartialEq, Eq)]
 pub struct Approval {
     owner: Key,
-    operator: Key,
+    spender: Key,
     token_id: String,
 }
 
 impl Approval {
-    pub fn new(owner: Key, operator: Key, token_id: TokenIdentifier) -> Self {
+    pub fn new(owner: Key, spender: Key, token_id: TokenIdentifier) -> Self {
         Self {
             owner,
-            operator,
+            spender,
             token_id: token_id.to_string(),
         }
     }
 }
 
 #[derive(Event, Debug, PartialEq, Eq)]
-pub struct ApprovalForAll {
+pub struct ApprovalRevoked {
     owner: Key,
-    operator: Option<Key>,
-    token_ids: Vec<String>,
+    token_id: String,
 }
 
-impl ApprovalForAll {
-    pub fn new(owner: Key, operator: Option<Key>, token_ids: Vec<TokenIdentifier>) -> Self {
+impl ApprovalRevoked {
+    pub fn new(owner: Key, token_id: TokenIdentifier) -> Self {
         Self {
             owner,
-            operator,
-            token_ids: token_ids
-                .iter()
-                .map(|token_id| token_id.to_string())
-                .collect(),
+            token_id: token_id.to_string(),
         }
     }
 }
 
 #[derive(Event, Debug, PartialEq, Eq)]
-pub struct Transfer {
+pub struct ApprovalForAll {
     owner: Key,
     operator: Option<Key>,
+}
```

Comment:
Since, the `ApprovalForAll` event doesn't have its `RevokeForAll` counterpart then a boolean flag should be added to indicate the action. I would call it `is_approved`.

Or should it be treated as always `true` and the `RevokeForAll` will be added later?

Reply:
Hi, Please check that commit https://github.com/casper-ecosystem/cep-78-enhanced-nft/pull/181/commits/59ea32601f19c61e4a27a2aafcc0c48ef221034e, please let me know if I missed anything on this new event thanks

Reply:
https://github.com/casper-ecosystem/cep-78-enhanced-nft/pull/181/commits/7d6de704544f8b8c9aa230f105ebf55ce10de39f

Reply:
Right now I have `RevokedForAll` which is not alike `ApprovalRevoked` nomenclature. Shall I rename to `RevokedForAll` to `ApprovalForAllRevoked` ?

Reply:
Let's have `RevokedForAll`, it's provides some symmetry with `ApprovalForAll`